

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  
  <title>Genomic Analysis &mdash; Python Outbreak.info API 0.1 documentation</title>
  

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  <link rel="stylesheet" href="_static/fonts.css" type="text/css" />

  
  

  
  

  

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Working with epidemiological data" href="epidem_analysis.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> Python Outbreak.info API
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul>
<li class="toctree-l1"><a class="reference internal" href="Cases_by_location.html">cases_by_location(location)</a></li>
<li class="toctree-l1"><a class="reference internal" href="Lineage_mutations.html">lineage_mutations(pango_lin, mutation, freq)</a></li>
<li class="toctree-l1"><a class="reference internal" href="Prevalence_by_location.html">prevalence_by_location(location, startswith)</a></li>
<li class="toctree-l1"><a class="reference internal" href="Global_Prev.html">global_prevalence(pango_lin, mutations, cumulative)</a></li>
<li class="toctree-l1"><a class="reference internal" href="Seq_counts.html">sequence_counts(location, cumulative, sub_admin)</a></li>
<li class="toctree-l1"><a class="reference internal" href="Mut_By_Lin.html">mutations_by_lineage(mutation, location, pango_lin)</a></li>
<li class="toctree-l1"><a class="reference internal" href="auth_setup.html">authenticate_new_user()</a></li>
<li class="toctree-l1"><a class="reference internal" href="daily_prev.html">daily_prev(pango_lin, location, mutations, cumulative)</a></li>
<li class="toctree-l1"><a class="reference internal" href="collection_date.html">collection_date(pango_lin, mutations=None, location=None)</a></li>
<li class="toctree-l1"><a class="reference internal" href="mutation_details.html">mutation_details(mutations)</a></li>
<li class="toctree-l1"><a class="reference internal" href="daily_lag.html">daily_lag(location=None)</a></li>
<li class="toctree-l1"><a class="reference internal" href="Outbreak_data_Functions.html">get_outbreak_data()</a></li>
</ul>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="epidem_analysis.html">Epidemiological data analyses</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Lineage prevalence analyses</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Python Outbreak.info API</a>
        
      </nav>


      <div class="wy-nav-content">
<div class="git-ribbon">
  <a href="http://github.com/SwissDataScienceCenter" rel="me">Join us on GitHub</a>
</div>

        
        <div class="rst-content">
        
          

















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
        
      <li>Genomic Analysis</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="_sources/genomic_analysis.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <section id="genomic-analysis">
<h1>Genomic Analysis<a class="headerlink" href="#genomic-analysis" title="Permalink to this headline">¶</a></h1>
<p>The <code class="docutils literal notranslate"><span class="pre">outbreak_data</span></code> package has contains multiple tools that can collect the relevant genomic information on SARS-CoV-2. Using a combination of these endpoints will allow you to get the information on variants of concern (VoC) and the current mutations to conduct your own multi-level genomic analysis on SARS-CoV-2. On this page, you’ll find a few example workflows that demonstrate data manipulation and visualization steps to track the progression of SARS-CoV-2.</p>
<p>Here is an example workflow that allows the user to manipulate the data to find all the XBB lineages prevalent in India within a 1-year timeframe:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># Perform authentication if you haven&#39;t already</span>
<span class="kn">from</span> <span class="nn">outbreak_data</span> <span class="kn">import</span> <span class="n">authenticate_user</span>
<span class="n">authenticate_user</span><span class="o">.</span><span class="n">authenticate_new_user</span><span class="p">()</span>

<span class="c1"># Import outbreak_data package</span>
<span class="kn">from</span> <span class="nn">outbreak_data</span> <span class="kn">import</span> <span class="n">outbreak_data</span> <span class="k">as</span> <span class="n">od</span>

<span class="c1"># Get the prevalence of all circulating XBB lineages in India</span>
<span class="n">data</span> <span class="o">=</span> <span class="n">od</span><span class="o">.</span><span class="n">prevalence_by_location</span><span class="p">(</span><span class="s2">&quot;IND&quot;</span><span class="p">,</span> <span class="n">startswith</span> <span class="o">=</span> <span class="s1">&#39;xbb&#39;</span><span class="p">)</span>
<span class="c1"># multiply prevalence values by 100% for scale</span>
<span class="n">data</span><span class="p">[</span><span class="s1">&#39;prevalence_rolling&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">data</span><span class="p">[</span><span class="s1">&#39;prevalence_rolling&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="n">x</span><span class="o">*</span><span class="mi">100</span><span class="p">)</span>
<span class="c1"># Search for data based on date range</span>
<span class="n">data</span> <span class="o">=</span> <span class="n">data</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s2">&quot;date&quot;</span><span class="p">)</span>
<span class="n">data</span> <span class="o">=</span> <span class="n">data</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">data</span><span class="p">[</span><span class="s2">&quot;date&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">between</span><span class="p">(</span><span class="s2">&quot;2020-09-12&quot;</span><span class="p">,</span> <span class="s2">&quot;2022-03-31&quot;</span><span class="p">)]</span>

<span class="c1">## Use the visual package of your choice to create an area graph using your data</span>
<span class="kn">import</span> <span class="nn">altair</span> <span class="k">as</span> <span class="nn">alt</span>

<span class="c1"># Graph of results</span>
<span class="n">alt</span><span class="o">.</span><span class="n">Chart</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="n">title</span> <span class="o">=</span> <span class="s2">&quot;Lineage Prevalence in India&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">mark_area</span><span class="p">()</span><span class="o">.</span><span class="n">encode</span><span class="p">(</span>
<span class="n">x</span><span class="o">=</span><span class="s1">&#39;date:T&#39;</span><span class="p">,</span>
<span class="n">y</span><span class="o">=</span><span class="n">alt</span><span class="o">.</span><span class="n">Y</span><span class="p">(</span><span class="s1">&#39;prevalence_rolling:Q&#39;</span><span class="p">),</span>
<span class="n">color</span> <span class="o">=</span> <span class="s1">&#39;lineage:N&#39;</span><span class="p">)</span>
</pre></div>
</div>
<div class="literal-block-wrapper docutils container" id="id1">
<div class="code-block-caption"><span class="caption-text">Output:</span><a class="headerlink" href="#id1" title="Permalink to this code">¶</a></div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>            <span class="n">date</span>  <span class="n">total_count</span>  <span class="n">lineage_count</span>     <span class="n">lineage</span>  <span class="n">prevalence</span>  \
 <span class="mi">3014</span> <span class="mi">2022</span><span class="o">-</span><span class="mi">09</span><span class="o">-</span><span class="mi">12</span>            <span class="mi">0</span>              <span class="mi">0</span>    <span class="n">xbb</span><span class="mf">.1.16</span>    <span class="mf">0.000000</span>
 <span class="mi">3781</span> <span class="mi">2022</span><span class="o">-</span><span class="mi">09</span><span class="o">-</span><span class="mi">12</span>            <span class="mi">0</span>              <span class="mi">0</span>     <span class="n">xbb</span><span class="mf">.2.3</span>    <span class="mf">0.000000</span>
 <span class="mi">2593</span> <span class="mi">2022</span><span class="o">-</span><span class="mi">09</span><span class="o">-</span><span class="mi">12</span>          <span class="mi">152</span>              <span class="mi">2</span>       <span class="n">xbb</span><span class="mf">.1</span>    <span class="mf">0.013158</span>
 <span class="mi">3782</span> <span class="mi">2022</span><span class="o">-</span><span class="mi">09</span><span class="o">-</span><span class="mi">13</span>            <span class="mi">0</span>              <span class="mi">0</span>     <span class="n">xbb</span><span class="mf">.2.3</span>    <span class="mf">0.000000</span>
 <span class="mi">3015</span> <span class="mi">2022</span><span class="o">-</span><span class="mi">09</span><span class="o">-</span><span class="mi">13</span>            <span class="mi">0</span>              <span class="mi">0</span>    <span class="n">xbb</span><span class="mf">.1.16</span>    <span class="mf">0.000000</span>
 <span class="o">...</span>         <span class="o">...</span>          <span class="o">...</span>            <span class="o">...</span>         <span class="o">...</span>         <span class="o">...</span>
 <span class="mi">4086</span> <span class="mi">2023</span><span class="o">-</span><span class="mi">03</span><span class="o">-</span><span class="mi">31</span>          <span class="mi">196</span>              <span class="mi">2</span>   <span class="n">xbb</span><span class="mf">.2.3.2</span>    <span class="mf">0.010204</span>
 <span class="mi">3322</span> <span class="mi">2023</span><span class="o">-</span><span class="mi">03</span><span class="o">-</span><span class="mi">31</span>          <span class="mi">196</span>             <span class="mi">29</span>  <span class="n">xbb</span><span class="mf">.1.16.1</span>    <span class="mf">0.147959</span>
 <span class="mi">2793</span> <span class="mi">2023</span><span class="o">-</span><span class="mi">03</span><span class="o">-</span><span class="mi">31</span>          <span class="mi">196</span>              <span class="mi">1</span>       <span class="n">xbb</span><span class="mf">.1</span>    <span class="mf">0.005102</span>
 <span class="mi">3381</span> <span class="mi">2023</span><span class="o">-</span><span class="mi">03</span><span class="o">-</span><span class="mi">31</span>          <span class="mi">196</span>              <span class="mi">7</span>  <span class="n">xbb</span><span class="mf">.1.16.2</span>    <span class="mf">0.035714</span>
 <span class="mi">3981</span> <span class="mi">2023</span><span class="o">-</span><span class="mi">03</span><span class="o">-</span><span class="mi">31</span>          <span class="mi">196</span>             <span class="mi">15</span>     <span class="n">xbb</span><span class="mf">.2.3</span>    <span class="mf">0.076531</span>

      <span class="n">prevalence_rolling</span>
 <span class="mi">3014</span>            <span class="mf">0.000000</span>
 <span class="mi">3781</span>            <span class="mf">0.000000</span>
 <span class="mi">2593</span>            <span class="mf">0.003451</span>
 <span class="mi">3782</span>            <span class="mf">0.000000</span>
 <span class="mi">3015</span>            <span class="mf">0.000000</span>
 <span class="o">...</span>                  <span class="o">...</span>
 <span class="mi">4086</span>            <span class="mf">0.031184</span>
 <span class="mi">3322</span>            <span class="mf">0.144578</span>
 <span class="mi">2793</span>            <span class="mf">0.014174</span>
 <span class="mi">3381</span>            <span class="mf">0.045358</span>
 <span class="mi">3981</span>            <span class="mf">0.084337</span>
<span class="p">[</span><span class="mi">985</span> <span class="n">rows</span> <span class="n">x</span> <span class="mi">6</span> <span class="n">columns</span><span class="p">]</span>
</pre></div>
</div>
</div>
<img alt="_images/prev_visual.png" src="_images/prev_visual.png" />
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The <a class="reference external" href="https://altair-viz.github.io/index.html">Vega-Altair</a> visualization package is used for demonstration purposes.         However, any Python visual package can be used to create graphical representations of the data.</p>
</div>
<p><strong>Lineage_Mutations Heatmap</strong></p>
<p>A basic but important question: how do we define a lineage? What mutations consistently appear in most sequences within a lineage? We could start by plotting some of the characteristic mutations of XBB occurring in 80% of sequences. Even better, we can do this for mulitple related lineages and compare them using a heatmap:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># Collecting 15 characteristic mutations and their prevalences for each variant</span>
<span class="n">lin0</span> <span class="o">=</span> <span class="n">od</span><span class="o">.</span><span class="n">lineage_mutations</span><span class="p">(</span><span class="s2">&quot;xbb&quot;</span><span class="p">,</span> <span class="n">freq</span> <span class="o">=</span> <span class="mf">0.80</span><span class="p">)</span>
<span class="n">lin0</span><span class="p">[</span><span class="s2">&quot;lineage&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="s2">&quot;xbb&quot;</span>
<span class="n">lin0</span> <span class="o">=</span> <span class="n">lin0</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="mi">15</span><span class="p">]</span>

<span class="n">lin1</span> <span class="o">=</span> <span class="n">od</span><span class="o">.</span><span class="n">lineage_mutations</span><span class="p">(</span><span class="s2">&quot;xbb.1&quot;</span><span class="p">,</span> <span class="n">freq</span> <span class="o">=</span> <span class="mf">0.80</span><span class="p">)</span>
<span class="n">lin1</span><span class="p">[</span><span class="s2">&quot;lineage&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="s2">&quot;xbb.1&quot;</span>
<span class="n">lin1</span> <span class="o">=</span> <span class="n">lin1</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="mi">15</span><span class="p">]</span>

<span class="n">lin2</span> <span class="o">=</span> <span class="n">od</span><span class="o">.</span><span class="n">lineage_mutations</span><span class="p">(</span><span class="s2">&quot;xbb.1.5&quot;</span><span class="p">,</span> <span class="n">freq</span> <span class="o">=</span> <span class="mf">0.80</span><span class="p">)</span>
<span class="n">lin2</span><span class="p">[</span><span class="s2">&quot;lineage&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="s2">&quot;xbb.1.5&quot;</span>
<span class="n">lin2</span> <span class="o">=</span> <span class="n">lin2</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="mi">15</span><span class="p">]</span>

<span class="n">lin3</span> <span class="o">=</span> <span class="n">od</span><span class="o">.</span><span class="n">lineage_mutations</span><span class="p">(</span><span class="s2">&quot;xbb.1.16&quot;</span><span class="p">,</span> <span class="n">freq</span> <span class="o">=</span> <span class="mf">0.80</span><span class="p">)</span>
<span class="n">lin3</span><span class="p">[</span><span class="s2">&quot;lineage&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="s2">&quot;xbb.1.16&quot;</span>
<span class="n">lin3</span> <span class="o">=</span> <span class="n">lin3</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="mi">15</span><span class="p">]</span>

<span class="n">lin4</span> <span class="o">=</span> <span class="n">od</span><span class="o">.</span><span class="n">lineage_mutations</span><span class="p">(</span><span class="s2">&quot;xbb.1.9&quot;</span><span class="p">,</span> <span class="n">freq</span> <span class="o">=</span> <span class="mf">0.80</span><span class="p">)</span>
<span class="n">lin4</span><span class="p">[</span><span class="s2">&quot;lineage&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="s2">&quot;xbb.1.9&quot;</span>
<span class="n">lin4</span> <span class="o">=</span> <span class="n">lin4</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="mi">15</span><span class="p">]</span>

<span class="n">lin5</span> <span class="o">=</span> <span class="n">od</span><span class="o">.</span><span class="n">lineage_mutations</span><span class="p">(</span><span class="s2">&quot;xbb.2.3&quot;</span><span class="p">,</span> <span class="n">freq</span> <span class="o">=</span> <span class="mf">0.80</span><span class="p">)</span>
<span class="n">lin5</span><span class="p">[</span><span class="s2">&quot;lineage&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="s2">&quot;xbb.2.3&quot;</span>
<span class="n">lin5</span> <span class="o">=</span> <span class="n">lin5</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="mi">15</span><span class="p">]</span>

<span class="c1"># Link each of the findings together for plotting</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">lin0</span><span class="p">,</span> <span class="n">lin1</span><span class="p">,</span> <span class="n">lin2</span><span class="p">,</span> <span class="n">lin3</span><span class="p">,</span> <span class="n">lin4</span><span class="p">,</span> <span class="n">lin5</span><span class="p">])</span>
<span class="n">df</span><span class="p">[</span><span class="s2">&quot;prevalence&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="s2">&quot;prevalence&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="n">x</span><span class="o">*</span><span class="mi">100</span><span class="p">)</span>
<span class="n">df</span><span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;prevalence&#39;</span><span class="p">:</span> <span class="s1">&#39;prevalence %&#39;</span><span class="p">})</span>

<span class="c1"># Plot mutation heatmap</span>
<span class="kn">import</span> <span class="nn">altair</span> <span class="k">as</span> <span class="nn">alt</span>

<span class="n">alt</span><span class="o">.</span><span class="n">Chart</span><span class="p">(</span><span class="n">df</span><span class="p">)</span><span class="o">.</span><span class="n">mark_rect</span><span class="p">()</span><span class="o">.</span><span class="n">encode</span><span class="p">(</span>
    <span class="n">x</span> <span class="o">=</span> <span class="s2">&quot;mutation:N&quot;</span><span class="p">,</span>
    <span class="n">y</span> <span class="o">=</span> <span class="s2">&quot;lineage:N&quot;</span><span class="p">,</span>
    <span class="n">color</span> <span class="o">=</span> <span class="s1">&#39;prevalence %:Q&#39;</span><span class="p">)</span>
</pre></div>
</div>
<div class="literal-block-wrapper docutils container" id="id2">
<div class="code-block-caption"><span class="caption-text">Output</span><a class="headerlink" href="#id2" title="Permalink to this code">¶</a></div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>         <span class="n">mutation</span>  <span class="n">mutation_count</span>  <span class="n">lineage_count</span>  <span class="n">lineage</span>   <span class="n">gene</span> <span class="n">ref_aa</span> <span class="n">alt_aa</span>  \
 <span class="mi">0</span>      <span class="n">orf6</span><span class="p">:</span><span class="n">d61l</span>            <span class="mi">4198</span>           <span class="mi">4423</span>      <span class="n">xbb</span>   <span class="n">ORF6</span>      <span class="n">D</span>      <span class="n">L</span>
 <span class="mi">1</span>         <span class="n">s</span><span class="p">:</span><span class="n">l24s</span>            <span class="mi">3919</span>           <span class="mi">4423</span>      <span class="n">xbb</span>      <span class="n">S</span>      <span class="n">L</span>      <span class="n">S</span>
 <span class="mi">2</span>        <span class="n">n</span><span class="p">:</span><span class="n">r203k</span>            <span class="mi">4378</span>           <span class="mi">4423</span>      <span class="n">xbb</span>      <span class="n">N</span>      <span class="n">R</span>      <span class="n">K</span>
 <span class="mi">3</span>        <span class="n">s</span><span class="p">:</span><span class="n">g339h</span>            <span class="mi">4054</span>           <span class="mi">4423</span>      <span class="n">xbb</span>      <span class="n">S</span>      <span class="n">G</span>      <span class="n">H</span>
 <span class="mi">4</span>        <span class="n">s</span><span class="p">:</span><span class="n">v445p</span>            <span class="mi">3643</span>           <span class="mi">4423</span>      <span class="n">xbb</span>      <span class="n">S</span>      <span class="n">V</span>      <span class="n">P</span>
 <span class="o">...</span>           <span class="o">...</span>             <span class="o">...</span>            <span class="o">...</span>      <span class="o">...</span>    <span class="o">...</span>    <span class="o">...</span>    <span class="o">...</span>
 <span class="mi">10</span>  <span class="n">orf1a</span><span class="p">:</span><span class="n">p3395h</span>            <span class="mi">3263</span>           <span class="mi">3279</span>  <span class="n">xbb</span><span class="mf">.2.3</span>  <span class="n">ORF1a</span>      <span class="n">P</span>      <span class="n">H</span>
 <span class="mi">11</span>       <span class="n">s</span><span class="p">:</span><span class="n">n969k</span>            <span class="mi">3261</span>           <span class="mi">3279</span>  <span class="n">xbb</span><span class="mf">.2.3</span>      <span class="n">S</span>      <span class="n">N</span>      <span class="n">K</span>
 <span class="mi">12</span>  <span class="n">orf1a</span><span class="p">:</span><span class="n">g1307s</span>            <span class="mi">3257</span>           <span class="mi">3279</span>  <span class="n">xbb</span><span class="mf">.2.3</span>  <span class="n">ORF1a</span>      <span class="n">G</span>      <span class="n">S</span>
 <span class="mi">13</span>       <span class="n">s</span><span class="p">:</span><span class="n">d614g</span>            <span class="mi">3257</span>           <span class="mi">3279</span>  <span class="n">xbb</span><span class="mf">.2.3</span>      <span class="n">S</span>      <span class="n">D</span>      <span class="n">G</span>
 <span class="mi">14</span>       <span class="n">s</span><span class="p">:</span><span class="n">q954h</span>            <span class="mi">3257</span>           <span class="mi">3279</span>  <span class="n">xbb</span><span class="mf">.2.3</span>      <span class="n">S</span>      <span class="n">Q</span>      <span class="n">H</span>

     <span class="n">codon_num</span> <span class="n">codon_end</span>          <span class="nb">type</span>  <span class="n">prevalence</span> <span class="o">%</span> <span class="n">change_length_nt</span>
 <span class="mi">0</span>          <span class="mi">61</span>      <span class="kc">None</span>  <span class="n">substitution</span>     <span class="mf">94.912955</span>             <span class="kc">None</span>
 <span class="mi">1</span>          <span class="mi">24</span>      <span class="kc">None</span>  <span class="n">substitution</span>     <span class="mf">88.605019</span>             <span class="kc">None</span>
 <span class="mi">2</span>         <span class="mi">203</span>      <span class="kc">None</span>  <span class="n">substitution</span>     <span class="mf">98.982591</span>             <span class="kc">None</span>
 <span class="mi">3</span>         <span class="mi">339</span>      <span class="kc">None</span>  <span class="n">substitution</span>     <span class="mf">91.657246</span>             <span class="kc">None</span>
 <span class="mi">4</span>         <span class="mi">445</span>      <span class="kc">None</span>  <span class="n">substitution</span>     <span class="mf">82.364911</span>             <span class="kc">None</span>
 <span class="o">...</span>        <span class="o">...</span>       <span class="o">...</span>           <span class="o">...</span>           <span class="o">...</span>              <span class="o">...</span>
 <span class="mi">10</span>       <span class="mi">3395</span>      <span class="kc">None</span>  <span class="n">substitution</span>     <span class="mf">99.512046</span>             <span class="kc">None</span>
 <span class="mi">11</span>        <span class="mi">969</span>      <span class="kc">None</span>  <span class="n">substitution</span>     <span class="mf">99.451052</span>             <span class="kc">None</span>
 <span class="mi">12</span>       <span class="mi">1307</span>      <span class="kc">None</span>  <span class="n">substitution</span>     <span class="mf">99.329064</span>             <span class="kc">None</span>
 <span class="mi">13</span>        <span class="mi">614</span>      <span class="kc">None</span>  <span class="n">substitution</span>     <span class="mf">99.329064</span>             <span class="kc">None</span>
 <span class="mi">14</span>        <span class="mi">954</span>      <span class="kc">None</span>  <span class="n">substitution</span>     <span class="mf">99.329064</span>             <span class="kc">None</span>
</pre></div>
</div>
</div>
<p>[90 rows x 12 columns]</p>
<img alt="_images/mut_by_lin.png" src="_images/mut_by_lin.png" />
<p><strong>Comparing Lineages</strong></p>
<p>To better understand the evolution of SARS-CoV-2, we may also want to know what mutations are shared between different viral lineages and how many shared mutations there are.</p>
<p>For this example we’ll look at three different lineages: B.1.1.7, P.1, and B.1.1.529. First, let’s get an idea of which mutations are shared between all three lineages:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">lin1</span> <span class="o">=</span> <span class="n">od</span><span class="o">.</span><span class="n">lineage_mutations</span><span class="p">(</span><span class="s2">&quot;b.1.1.7&quot;</span><span class="p">)</span>
<span class="n">lin1</span><span class="p">[</span><span class="s2">&quot;lineage&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="s2">&quot;b.1.1.7&quot;</span>

<span class="n">lin2</span> <span class="o">=</span> <span class="n">od</span><span class="o">.</span><span class="n">lineage_mutations</span><span class="p">(</span><span class="s2">&quot;P.1&quot;</span><span class="p">)</span>
<span class="n">lin2</span><span class="p">[</span><span class="s2">&quot;lineage&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="s2">&quot;P.1&quot;</span>

<span class="n">lin3</span> <span class="o">=</span> <span class="n">od</span><span class="o">.</span><span class="n">lineage_mutations</span><span class="p">(</span><span class="s2">&quot;b.1.1.529&quot;</span><span class="p">)</span>
<span class="n">lin3</span><span class="p">[</span><span class="s2">&quot;lineage&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="s2">&quot;b.1.1.529&quot;</span>

<span class="c1">#Finding intersections</span>

<span class="c1">#Created an optional but convenient intersecting function</span>
<span class="k">def</span> <span class="nf">intersect</span><span class="p">(</span><span class="n">lst1</span><span class="p">,</span> <span class="n">lst2</span><span class="p">):</span>
    <span class="k">return</span> <span class="nb">set</span><span class="p">(</span><span class="n">lst1</span><span class="p">)</span><span class="o">.</span><span class="n">intersection</span><span class="p">(</span><span class="n">lst2</span><span class="p">)</span>

<span class="n">a</span> <span class="o">=</span> <span class="nb">set</span><span class="p">(</span><span class="n">lin1</span><span class="p">[</span><span class="s1">&#39;mutation&#39;</span><span class="p">])</span>
<span class="n">b</span> <span class="o">=</span> <span class="nb">set</span><span class="p">(</span><span class="n">lin2</span><span class="p">[</span><span class="s1">&#39;mutation&#39;</span><span class="p">])</span>
<span class="n">c</span> <span class="o">=</span> <span class="nb">set</span><span class="p">(</span><span class="n">lin3</span><span class="p">[</span><span class="s1">&#39;mutation&#39;</span><span class="p">])</span>

<span class="n">m12</span> <span class="o">=</span> <span class="n">intersect</span><span class="p">(</span><span class="n">a</span><span class="p">,</span> <span class="n">b</span><span class="p">)</span>
<span class="n">m13</span> <span class="o">=</span> <span class="n">intersect</span><span class="p">(</span><span class="n">a</span><span class="p">,</span> <span class="n">c</span><span class="p">)</span>
<span class="n">m23</span> <span class="o">=</span> <span class="n">intersect</span><span class="p">(</span><span class="n">b</span><span class="p">,</span> <span class="n">c</span><span class="p">)</span>
<span class="n">m123</span> <span class="o">=</span> <span class="n">intersect</span><span class="p">(</span><span class="n">m12</span><span class="p">,</span> <span class="n">c</span><span class="p">)</span>

<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;Set 1.2:&#39;</span><span class="p">,</span> <span class="n">m12</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;Set 1.3:&#39;</span><span class="p">,</span> <span class="n">m13</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;Set 2.3:&#39;</span> <span class="p">,</span> <span class="n">m23</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;Set 1.2.3:&#39;</span><span class="p">,</span> <span class="n">m123</span><span class="p">)</span>
</pre></div>
</div>
<div class="literal-block-wrapper docutils container" id="id3">
<div class="code-block-caption"><span class="caption-text">Output</span><a class="headerlink" href="#id3" title="Permalink to this code">¶</a></div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span> <span class="n">Set</span> <span class="mf">1.2</span><span class="p">:</span> <span class="p">{</span><span class="s1">&#39;orf1b:p314l&#39;</span><span class="p">,</span> <span class="s1">&#39;orf8:s84l&#39;</span><span class="p">,</span> <span class="s1">&#39;orf1a:del3675/3677&#39;</span><span class="p">,</span> <span class="s1">&#39;n:r203k&#39;</span><span class="p">,</span> <span class="s1">&#39;s:d614g&#39;</span><span class="p">,</span> <span class="s1">&#39;s:n501y&#39;</span><span class="p">,</span> <span class="s1">&#39;n:g204r&#39;</span><span class="p">}</span>
 <span class="n">Set</span> <span class="mf">1.3</span><span class="p">:</span> <span class="p">{</span><span class="s1">&#39;s:p681h&#39;</span><span class="p">,</span> <span class="s1">&#39;orf1b:p314l&#39;</span><span class="p">,</span> <span class="s1">&#39;orf8:s84l&#39;</span><span class="p">,</span> <span class="s1">&#39;n:r203k&#39;</span><span class="p">,</span> <span class="s1">&#39;s:d614g&#39;</span><span class="p">,</span> <span class="s1">&#39;n:g204r&#39;</span><span class="p">}</span>
 <span class="n">Set</span> <span class="mf">2.3</span><span class="p">:</span> <span class="p">{</span><span class="s1">&#39;orf1b:p314l&#39;</span><span class="p">,</span> <span class="s1">&#39;s:h655y&#39;</span><span class="p">,</span> <span class="s1">&#39;orf8:s84l&#39;</span><span class="p">,</span> <span class="s1">&#39;n:r203k&#39;</span><span class="p">,</span> <span class="s1">&#39;s:d614g&#39;</span><span class="p">,</span> <span class="s1">&#39;n:g204r&#39;</span><span class="p">}</span>
 <span class="n">Set</span> <span class="mf">1.2.3</span><span class="p">:</span> <span class="p">{</span><span class="s1">&#39;orf1b:p314l&#39;</span><span class="p">,</span> <span class="s1">&#39;orf8:s84l&#39;</span><span class="p">,</span> <span class="s1">&#39;n:r203k&#39;</span><span class="p">,</span> <span class="s1">&#39;s:d614g&#39;</span><span class="p">,</span> <span class="s1">&#39;n:g204r&#39;</span><span class="p">}</span>
</pre></div>
</div>
</div>
<p>Finally, we can make a visual based on our findings and create a Venn diagram of our results:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># Import libraries for venn diagram visual</span>
<span class="kn">from</span> <span class="nn">matplotlib_venn</span> <span class="kn">import</span> <span class="n">venn3</span><span class="p">,</span> <span class="n">venn3_circles</span>
<span class="kn">from</span> <span class="nn">matplotlib</span> <span class="kn">import</span> <span class="n">pyplot</span> <span class="k">as</span> <span class="n">plt</span>

<span class="c1"># Using matplotlib_venn to make venn diagram with custom visuals</span>
 <span class="c1">## The venn diagram package used here automatically determines the number</span>
 <span class="c1">## of intersecting values and plots accordingly</span>

<span class="n">vd3</span> <span class="o">=</span> <span class="n">venn3</span><span class="p">([</span><span class="n">a</span><span class="p">,</span> <span class="n">b</span><span class="p">,</span> <span class="n">c</span><span class="p">],</span> <span class="n">set_labels</span> <span class="o">=</span> <span class="p">(</span><span class="s1">&#39;Alpha/B.1.1.7&#39;</span><span class="p">,</span> <span class="s1">&#39;Gamma/P.1&#39;</span><span class="p">,</span> <span class="s1">&#39;Omicron/B.1.1.529&#39;</span><span class="p">),</span> <span class="n">set_colors</span><span class="o">=</span><span class="p">(</span><span class="s1">&#39;#c4e6ff&#39;</span><span class="p">,</span> <span class="s1">&#39;#F4ACB7&#39;</span><span class="p">,</span><span class="s1">&#39;#9D8189&#39;</span><span class="p">),</span>
 <span class="n">alpha</span> <span class="o">=</span> <span class="mf">0.8</span><span class="p">)</span>
<span class="n">c</span> <span class="o">=</span> <span class="n">venn3_circles</span><span class="p">([</span><span class="n">a</span><span class="p">,</span> <span class="n">b</span><span class="p">,</span> <span class="n">c</span><span class="p">,],</span> <span class="n">linestyle</span> <span class="o">=</span> <span class="s1">&#39;-.&#39;</span><span class="p">,</span> <span class="n">linewidth</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s1">&#39;grey&#39;</span><span class="p">)</span>
<span class="k">for</span> <span class="n">text</span> <span class="ow">in</span> <span class="n">vd3</span><span class="o">.</span><span class="n">set_labels</span><span class="p">:</span>
 <span class="n">text</span><span class="o">.</span><span class="n">set_fontsize</span><span class="p">(</span><span class="mi">16</span><span class="p">);</span>
<span class="k">for</span> <span class="n">text</span> <span class="ow">in</span> <span class="n">vd3</span><span class="o">.</span><span class="n">subset_labels</span><span class="p">:</span>
 <span class="n">text</span><span class="o">.</span><span class="n">set_fontsize</span><span class="p">(</span><span class="mi">16</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">title</span><span class="p">(</span><span class="s1">&#39;Number of Overlapping Mutations in 3 Variants&#39;</span><span class="p">,</span><span class="n">fontname</span> <span class="o">=</span><span class="s1">&#39;Helvetica&#39;</span><span class="p">,</span><span class="n">fontweight</span> <span class="o">=</span><span class="s1">&#39;bold&#39;</span><span class="p">,</span><span class="n">fontsize</span> <span class="o">=</span> <span class="mi">18</span><span class="p">,</span>
 <span class="n">pad</span> <span class="o">=</span> <span class="mi">30</span><span class="p">,</span><span class="n">backgroundcolor</span> <span class="o">=</span><span class="s1">&#39;#cbe7e3&#39;</span><span class="p">,</span><span class="n">color</span> <span class="o">=</span><span class="s1">&#39;black&#39;</span><span class="p">,</span> <span class="n">style</span> <span class="o">=</span><span class="s1">&#39;normal&#39;</span><span class="p">);</span>
<span class="n">c</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_lw</span><span class="p">(</span><span class="mf">7.0</span><span class="p">)</span>
<span class="n">c</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_ls</span><span class="p">(</span><span class="s1">&#39;:&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
<img alt="_images/compare.png" src="_images/compare.png" />
<p>We can see that the graph matches the number of intersecting mutations that we previously saw, given that the venn diagram also takes into account unique values for each set.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Please see the <a class="reference external" href="https://python-graph-gallery.com/venn-diagram/">matplotlib venn diagram documentation</a> and <a class="reference external" href="https://towardsdatascience.com/how-to-create-and-beautify-venn-diagrams-in-python-331129bd4ed3">this article</a>  for more info on how to create these graphs.</p>
</div>
<p><strong>Finding the Most Prevalent Lineages</strong></p>
<p>If we wanted to determine and plot the top four most prevalent lineages in India, we can make a few queries and use a few simple commands to create a table that shows us what these lineages are:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">data</span><span class="o">=</span><span class="n">od</span><span class="o">.</span><span class="n">prevalence_by_location</span><span class="p">(</span><span class="s2">&quot;IND&quot;</span><span class="p">)</span>
<span class="n">most_prev</span> <span class="o">=</span> <span class="n">data</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="s1">&#39;lineage&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="nb">max</span><span class="p">)</span> <span class="c1"># Finds the lineages with the most hits</span>
<span class="n">most_prev</span> <span class="o">=</span> <span class="n">most_prev</span><span class="o">.</span><span class="n">mask</span><span class="p">(</span><span class="n">most_prev</span> <span class="o">==</span> <span class="s1">&#39;&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">dropna</span><span class="p">(</span><span class="n">how</span> <span class="o">=</span> <span class="s1">&#39;any&#39;</span><span class="p">)</span> <span class="c1"># Drop any unknowns</span>
<span class="n">most_prev</span> <span class="o">=</span> <span class="n">most_prev</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="mi">4</span><span class="p">]</span>
<span class="nb">print</span><span class="p">(</span><span class="n">most_prev</span><span class="p">)</span>
</pre></div>
</div>
<div class="literal-block-wrapper docutils container" id="id4">
<div class="code-block-caption"><span class="caption-text">Output</span><a class="headerlink" href="#id4" title="Permalink to this code">¶</a></div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>                  <span class="n">date</span>  <span class="n">total_count</span>  <span class="n">lineage_count</span>    <span class="n">lineage</span>  <span class="n">prevalence</span>  \
 <span class="n">lineage</span>
 <span class="n">ba</span><span class="mf">.2</span>       <span class="mi">2023</span><span class="o">-</span><span class="mi">04</span><span class="o">-</span><span class="mi">20</span>         <span class="mi">5668</span>           <span class="mi">1445</span>       <span class="n">ba</span><span class="mf">.2</span>    <span class="mf">0.822785</span>
 <span class="n">ba</span><span class="mf">.2.10.1</span>  <span class="mi">2023</span><span class="o">-</span><span class="mi">04</span><span class="o">-</span><span class="mi">19</span>         <span class="mi">5668</span>             <span class="mi">93</span>  <span class="n">ba</span><span class="mf">.2.10.1</span>    <span class="mf">0.285714</span>
 <span class="n">bq</span><span class="mf">.1.1</span>     <span class="mi">2023</span><span class="o">-</span><span class="mi">03</span><span class="o">-</span><span class="mi">27</span>          <span class="mi">402</span>              <span class="mi">7</span>     <span class="n">bq</span><span class="mf">.1.1</span>    <span class="mf">0.428571</span>
 <span class="n">ch</span><span class="mf">.1.1</span>     <span class="mi">2023</span><span class="o">-</span><span class="mi">02</span><span class="o">-</span><span class="mi">13</span>          <span class="mi">119</span>              <span class="mi">4</span>     <span class="n">ch</span><span class="mf">.1.1</span>    <span class="mf">0.400000</span>

            <span class="n">prevalence_rolling</span>
 <span class="n">lineage</span>
 <span class="n">ba</span><span class="mf">.2</span>                 <span class="mf">0.677541</span>
 <span class="n">ba</span><span class="mf">.2.10.1</span>            <span class="mf">0.095541</span>
 <span class="n">bq</span><span class="mf">.1.1</span>               <span class="mf">0.156863</span>
 <span class="n">ch</span><span class="mf">.1.1</span>               <span class="mf">0.066667</span>
</pre></div>
</div>
</div>
<p>Next we’ll collect the prevalence data on each of the four lineages:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># Retrieve the official data on the prevalences of these lineages using `daily_prev()`</span>
<span class="n">d1</span> <span class="o">=</span> <span class="n">od</span><span class="o">.</span><span class="n">daily_prev</span><span class="p">(</span><span class="s1">&#39;ba.2&#39;</span><span class="p">,</span> <span class="s2">&quot;IND&quot;</span><span class="p">)</span>
<span class="n">d2</span> <span class="o">=</span> <span class="n">od</span><span class="o">.</span><span class="n">daily_prev</span><span class="p">(</span><span class="s1">&#39;ba.2.10.1&#39;</span><span class="p">,</span> <span class="s2">&quot;IND&quot;</span><span class="p">)</span>
<span class="n">d3</span> <span class="o">=</span> <span class="n">od</span><span class="o">.</span><span class="n">daily_prev</span><span class="p">(</span><span class="s1">&#39;bq.1.1&#39;</span><span class="p">,</span> <span class="s2">&quot;IND&quot;</span><span class="p">)</span>
<span class="n">d4</span> <span class="o">=</span> <span class="n">od</span><span class="o">.</span><span class="n">daily_prev</span><span class="p">(</span> <span class="s1">&#39;ch.1.1&#39;</span><span class="p">,</span> <span class="s2">&quot;IND&quot;</span><span class="p">)</span>

<span class="c1"># Formatting for creating the graph</span>
<span class="n">d1</span><span class="p">[</span><span class="s1">&#39;lineage&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="s1">&#39;ba.2&#39;</span>
<span class="n">d2</span><span class="p">[</span><span class="s1">&#39;lineage&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="s1">&#39;ba.2.10.1&#39;</span>
<span class="n">d3</span><span class="p">[</span><span class="s1">&#39;lineage&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="s1">&#39;bq.1.1&#39;</span>
<span class="n">d4</span><span class="p">[</span><span class="s1">&#39;lineage&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="s1">&#39;ch.1.1&#39;</span>

<span class="c1"># Group together data from each lineage</span>
<span class="n">data</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">d1</span><span class="p">,</span> <span class="n">d2</span><span class="p">,</span> <span class="n">d3</span><span class="p">,</span> <span class="n">d4</span><span class="p">])</span>
<span class="n">data</span> <span class="o">=</span> <span class="n">data</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;proportion&#39;</span><span class="p">:</span> <span class="s1">&#39;proportion (%)&#39;</span><span class="p">})</span>

<span class="c1">#Pick a date range to analyze</span>
<span class="n">data</span> <span class="o">=</span> <span class="n">data</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s2">&quot;date&quot;</span><span class="p">)</span>
<span class="n">data</span> <span class="o">=</span> <span class="n">data</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">data</span><span class="p">[</span><span class="s2">&quot;date&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">between</span><span class="p">(</span><span class="s2">&quot;2022-09-12&quot;</span><span class="p">,</span> <span class="s2">&quot;2023-03-31&quot;</span><span class="p">)]</span>
<span class="c1"># Increase prevalence by 100%</span>
<span class="n">data</span><span class="p">[</span><span class="s1">&#39;proportion&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">data</span><span class="p">[</span><span class="s1">&#39;proportion&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="n">x</span><span class="o">*</span><span class="mi">100</span><span class="p">)</span>

<span class="c1">#Graph using preferred visual package</span>
<span class="kn">import</span> <span class="nn">altair</span> <span class="k">as</span> <span class="nn">alt</span>
<span class="n">alt</span><span class="o">.</span><span class="n">Chart</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="n">title</span> <span class="o">=</span> <span class="s2">&quot;Top 4 Most Prevalent Lineages in India&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">mark_area</span><span class="p">()</span><span class="o">.</span><span class="n">encode</span><span class="p">(</span>
<span class="n">x</span><span class="o">=</span><span class="s1">&#39;date:T&#39;</span><span class="p">,</span>
<span class="n">y</span><span class="o">=</span><span class="n">alt</span><span class="o">.</span><span class="n">Y</span><span class="p">(</span><span class="s1">&#39;proportion (%):Q&#39;</span><span class="p">),</span>
<span class="n">color</span> <span class="o">=</span> <span class="s1">&#39;lineage:N&#39;</span><span class="p">)</span>
</pre></div>
</div>
<img alt="_images/top4.png" src="_images/top4.png" />
</section>


           </div>
           
          </div>
          <footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
        <a href="epidem_analysis.html" class="btn btn-neutral float-left" title="Working with epidemiological data" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>
        &#169; Copyright .

    </p>
  </div>
    
    
    
    Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>
        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>