<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Welcome to the Python Outbreak.info package docs! &mdash; Python Outbreak.info API 0.1 documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="_static/custom.css" type="text/css" />
      <link rel="stylesheet" href="_static/fonts.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="_static/jquery.js?v=5d32c60e"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="_static/documentation_options.js?v=2709fde1"></script>
        <script src="_static/doctools.js?v=888ff710"></script>
        <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="authenticate_new_user()" href="auth_setup.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="#" class="icon icon-home">
            Python Outbreak.info API
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul>
<li class="toctree-l1"><a class="reference internal" href="auth_setup.html">authenticate_new_user()</a></li>
<li class="toctree-l1"><a class="reference internal" href="cases_by_location.html">cases_by_location(location)</a></li>
<li class="toctree-l1"><a class="reference internal" href="lineage_mutations.html">lineage_mutations(pango_lin, mutation, freq)</a></li>
<li class="toctree-l1"><a class="reference internal" href="all_lineage_prevalences.html">all_lineage_prevalences(location, startswith)</a></li>
<li class="toctree-l1"><a class="reference internal" href="global_prevalence.html">global_prevalence(pango_lin, mutations, cumulative, lineage_crumbs=False)</a></li>
<li class="toctree-l1"><a class="reference internal" href="seq_counts.html">sequence_counts(location, cumulative, sub_admin)</a></li>
<li class="toctree-l1"><a class="reference internal" href="mutations_by_lineage.html">mutations_by_lineage(mutation, location, pango_lin)</a></li>
<li class="toctree-l1"><a class="reference internal" href="prevalence_by_location.html">prevalence_by_location(pango_lin, location, mutations=None, datemin=None, datemax=None, cumulative=None, lineage_crumbs=False)</a></li>
<li class="toctree-l1"><a class="reference internal" href="lineage_by_sub_admin.html">lineage_by_sub_admin(pango_lin, mutations, location, ndays=0, detected)</a></li>
<li class="toctree-l1"><a class="reference internal" href="collection_date.html">collection_date(pango_lin, mutations=None, location=None)</a></li>
<li class="toctree-l1"><a class="reference internal" href="submission_date.html">submission_date(pango_lin, mutations=None, location)</a></li>
<li class="toctree-l1"><a class="reference internal" href="mutation_details.html">mutation_details(mutations)</a></li>
<li class="toctree-l1"><a class="reference internal" href="daily_lag.html">daily_lag(location=None)</a></li>
<li class="toctree-l1"><a class="reference internal" href="wildcard_lineage.html">wildcard_lineage(name)</a></li>
<li class="toctree-l1"><a class="reference internal" href="wildcard_location.html">wildcard_location(name)</a></li>
<li class="toctree-l1"><a class="reference internal" href="wildcard_mutations.html">wildcard_mutations(name)</a></li>
<li class="toctree-l1"><a class="reference internal" href="location_details.html">location_details(location)</a></li>
<li class="toctree-l1"><a class="reference internal" href="outbreak_data_functions.html">get_outbreak_data()</a></li>
<li class="toctree-l1"><a class="reference internal" href="growth_rates.html">growth_rates(lineage, location)</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="epidem_analysis.html">Epidemiological data analyses</a></li>
<li class="toctree-l1"><a class="reference internal" href="lineage_prevalence.html">Lineage Prevalence Analyses</a></li>
<li class="toctree-l1"><a class="reference internal" href="mutation_analysis.html">Mutation Data Analyses</a></li>
<li class="toctree-l1"><a class="reference internal" href="cryptic_vars.html">Dealing with Cryptic Variants</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="#">Python Outbreak.info API</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="#" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Welcome to the Python Outbreak.info package docs!</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/index.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="welcome-to-the-python-outbreak-info-package-docs">
<h1>Welcome to the Python Outbreak.info package docs!<a class="headerlink" href="#welcome-to-the-python-outbreak-info-package-docs" title="Link to this heading"></a></h1>
<p>Here you can find information on the functions you will use to collect and analyze SARS-COV-2  data from the Outbreak.info API.
Our package pulls data from the <a class="reference external" href="https://api.outbreak.info/">Outbreak.info API</a> and is reflected on our <a class="reference external" href="https://outbreak.info/">Outbreak.info web interface</a></p>
<section id="installation">
<h2>Installation<a class="headerlink" href="#installation" title="Link to this heading"></a></h2>
<p>We recommend installing the package via pip using:</p>
<p><code class="docutils literal notranslate"><span class="pre">pip</span> <span class="pre">install</span> <span class="pre">python-outbreak-info</span></code></p>
<p>Alternatively, the package can be directly installed from source via pip:</p>
<p><code class="docutils literal notranslate"><span class="pre">pip</span> <span class="pre">install</span> <span class="pre">git+https://github.com/outbreak-info/python-outbreak-info.git</span></code></p>
</section>
<section id="getting-started">
<h2>Getting Started<a class="headerlink" href="#getting-started" title="Link to this heading"></a></h2>
<p>The Python Outbreak.info package contains key functions for accessing genomic and epidemiological data for SARS-CoV-2. Access to genomic data requires logging in using GISAID credentials to generate an API key, using the <code class="docutils literal notranslate"><span class="pre">authenticate_new_user()</span></code> function. To perform authentication, you’ll need to first run</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">outbreak_data</span> <span class="kn">import</span> <span class="n">authenticate_user</span>
<span class="n">authenticate_user</span><span class="o">.</span><span class="n">authenticate_new_user</span><span class="p">()</span>
</pre></div>
</div>
<p>and then you should be able access all of the functionality of the package. Most of the rest of the tools are available within the <code class="docutils literal notranslate"><span class="pre">outbreak_data</span></code> component of the package. For example:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">outbreak_data</span> <span class="kn">import</span> <span class="n">outbreak_data</span>
<span class="n">lin_list</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;B.1.1.7&#39;</span><span class="p">,</span><span class="s1">&#39;B.1.351&#39;</span><span class="p">,</span><span class="s1">&#39;B.1.617.2&#39;</span><span class="p">]</span>
<span class="c1"># request lineages occurring with minimum frequency of 0.05 (5%)</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">outbreak_data</span><span class="o">.</span><span class="n">lineage_mutations</span><span class="p">(</span><span class="n">lin_list</span><span class="p">,</span><span class="n">freq</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="c1"># filter mutations and sort by codon number</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">df</span><span class="p">[</span><span class="s1">&#39;gene&#39;</span><span class="p">]</span><span class="o">==</span><span class="s1">&#39;S&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;codon_num&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>For location-specific analyses, users will need to supply the appropriate location code corresponding to their location of interest. To do this, we provide an ID lookup tool kit via the <code class="docutils literal notranslate"><span class="pre">outbreak_tools</span></code> part of the package. An example lookup should look like:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">outbreak_tools</span> <span class="kn">import</span> <span class="n">outbreak_tools</span>
<span class="n">location_list</span> <span class="o">=</span> <span class="n">outbreak_tools</span><span class="o">.</span><span class="n">id_lookup</span><span class="p">([</span><span class="s1">&#39;Illinois&#39;</span><span class="p">,</span><span class="s1">&#39;South Africa&#39;</span><span class="p">,</span><span class="s1">&#39;Chile&#39;</span><span class="p">])</span>
<span class="c1"># which returns [&#39;USA_US-IL&#39;, &#39;ZAF&#39;, &#39;CHL&#39;]</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>This project is under active development.</p>
</div>
</section>
<section id="core-outbreak-data-tools">
<h2>Core Outbreak Data Tools<a class="headerlink" href="#core-outbreak-data-tools" title="Link to this heading"></a></h2>
<div class="toctree-wrapper compound">
<ul>
<li class="toctree-l1"><a class="reference internal" href="auth_setup.html">authenticate_new_user()</a><ul>
<li class="toctree-l2"><a class="reference internal" href="auth_setup.html#authenticate_user.authenticate_new_user"><code class="docutils literal notranslate"><span class="pre">authenticate_new_user()</span></code></a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="cases_by_location.html">cases_by_location(location)</a><ul>
<li class="toctree-l2"><a class="reference internal" href="cases_by_location.html#outbreak_data.cases_by_location"><code class="docutils literal notranslate"><span class="pre">cases_by_location()</span></code></a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="lineage_mutations.html">lineage_mutations(pango_lin, mutation, freq)</a><ul>
<li class="toctree-l2"><a class="reference internal" href="lineage_mutations.html#outbreak_data.lineage_mutations"><code class="docutils literal notranslate"><span class="pre">lineage_mutations()</span></code></a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="all_lineage_prevalences.html">all_lineage_prevalences(location, startswith)</a><ul>
<li class="toctree-l2"><a class="reference internal" href="all_lineage_prevalences.html#outbreak_data.all_lineage_prevalences"><code class="docutils literal notranslate"><span class="pre">all_lineage_prevalences()</span></code></a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="global_prevalence.html">global_prevalence(pango_lin, mutations, cumulative, lineage_crumbs=False)</a><ul>
<li class="toctree-l2"><a class="reference internal" href="global_prevalence.html#outbreak_data.global_prevalence"><code class="docutils literal notranslate"><span class="pre">global_prevalence()</span></code></a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="seq_counts.html">sequence_counts(location, cumulative, sub_admin)</a><ul>
<li class="toctree-l2"><a class="reference internal" href="seq_counts.html#outbreak_data.sequence_counts"><code class="docutils literal notranslate"><span class="pre">sequence_counts()</span></code></a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="mutations_by_lineage.html">mutations_by_lineage(mutation, location, pango_lin)</a><ul>
<li class="toctree-l2"><a class="reference internal" href="mutations_by_lineage.html#outbreak_data.mutations_by_lineage"><code class="docutils literal notranslate"><span class="pre">mutations_by_lineage()</span></code></a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="prevalence_by_location.html">prevalence_by_location(pango_lin, location, mutations=None, datemin=None, datemax=None, cumulative=None, lineage_crumbs=False)</a><ul>
<li class="toctree-l2"><a class="reference internal" href="prevalence_by_location.html#outbreak_data.prevalence_by_location"><code class="docutils literal notranslate"><span class="pre">prevalence_by_location()</span></code></a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="lineage_by_sub_admin.html">lineage_by_sub_admin(pango_lin, mutations, location, ndays=0, detected)</a><ul>
<li class="toctree-l2"><a class="reference internal" href="lineage_by_sub_admin.html#outbreak_data.lineage_by_sub_admin"><code class="docutils literal notranslate"><span class="pre">lineage_by_sub_admin()</span></code></a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="collection_date.html">collection_date(pango_lin, mutations=None, location=None)</a><ul>
<li class="toctree-l2"><a class="reference internal" href="collection_date.html#outbreak_data.collection_date"><code class="docutils literal notranslate"><span class="pre">collection_date()</span></code></a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="submission_date.html">submission_date(pango_lin, mutations=None, location)</a><ul>
<li class="toctree-l2"><a class="reference internal" href="submission_date.html#outbreak_data.submission_date"><code class="docutils literal notranslate"><span class="pre">submission_date()</span></code></a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="mutation_details.html">mutation_details(mutations)</a><ul>
<li class="toctree-l2"><a class="reference internal" href="mutation_details.html#outbreak_data.mutation_details"><code class="docutils literal notranslate"><span class="pre">mutation_details()</span></code></a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="daily_lag.html">daily_lag(location=None)</a><ul>
<li class="toctree-l2"><a class="reference internal" href="daily_lag.html#outbreak_data.daily_lag"><code class="docutils literal notranslate"><span class="pre">daily_lag()</span></code></a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="wildcard_lineage.html">wildcard_lineage(name)</a><ul>
<li class="toctree-l2"><a class="reference internal" href="wildcard_lineage.html#outbreak_data.wildcard_lineage"><code class="docutils literal notranslate"><span class="pre">wildcard_lineage()</span></code></a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="wildcard_location.html">wildcard_location(name)</a><ul>
<li class="toctree-l2"><a class="reference internal" href="wildcard_location.html#outbreak_data.wildcard_location"><code class="docutils literal notranslate"><span class="pre">wildcard_location()</span></code></a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="wildcard_mutations.html">wildcard_mutations(name)</a><ul>
<li class="toctree-l2"><a class="reference internal" href="wildcard_mutations.html#outbreak_data.wildcard_mutations"><code class="docutils literal notranslate"><span class="pre">wildcard_mutations()</span></code></a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="location_details.html">location_details(location)</a><ul>
<li class="toctree-l2"><a class="reference internal" href="location_details.html#outbreak_data.location_details"><code class="docutils literal notranslate"><span class="pre">location_details()</span></code></a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="outbreak_data_functions.html">get_outbreak_data()</a><ul>
<li class="toctree-l2"><a class="reference internal" href="outbreak_data_functions.html#outbreak_data.get_outbreak_data"><code class="docutils literal notranslate"><span class="pre">get_outbreak_data()</span></code></a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="growth_rates.html">growth_rates(lineage, location)</a><ul>
<li class="toctree-l2"><a class="reference internal" href="growth_rates.html#outbreak_data.growth_rates"><code class="docutils literal notranslate"><span class="pre">growth_rates()</span></code></a></li>
</ul>
</li>
</ul>
</div>
</section>
<section id="example-analyses">
<h2>Example analyses:<a class="headerlink" href="#example-analyses" title="Link to this heading"></a></h2>
<div class="toctree-wrapper compound">
<ul>
<li class="toctree-l1"><a class="reference internal" href="epidem_analysis.html">Epidemiological data analyses</a></li>
<li class="toctree-l1"><a class="reference internal" href="lineage_prevalence.html">Lineage Prevalence Analyses</a></li>
<li class="toctree-l1"><a class="reference internal" href="mutation_analysis.html">Mutation Data Analyses</a></li>
<li class="toctree-l1"><a class="reference internal" href="cryptic_vars.html">Dealing with Cryptic Variants</a></li>
</ul>
</div>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="auth_setup.html" class="btn btn-neutral float-right" title="authenticate_new_user()" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright .</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>